packages/hardhat-etherscan/src/network/prober.ts

```diff
@@ -24,30 +24,33 @@ const networkIDtoEndpoint: NetworkMap = {
   [NetworkID.KOVAN]: "https://api-kovan.etherscan.io/api",
 };
 
-export class NetworkProberError extends HardhatPluginError {
-  constructor(message: string) {
-    super(pluginName, message);
-  }
-}
-
-export async function getEtherscanEndpoint(provider: EthereumProvider) {
+export async function getEtherscanEndpoint(
+  provider: EthereumProvider,
+  networkName: string
+): Promise<string> {
   const chainID = parseInt(await provider.send("eth_chainId"), 16) as NetworkID;
 
   const endpoint = networkIDtoEndpoint[chainID];
-  if (endpoint !== null && endpoint !== undefined) {
-    // Beware: this delays URL validation until it is effectively "used".
-    // Tests should take this into account.
-    return new URL(endpoint);
+
+  if (endpoint === null || endpoint === undefined) {
```

Comment:
```suggestion
  if (endpoint === undefined) {
```

this can't be `null`